src/pip/_internal/req/req_install.py

```diff
@@ -250,14 +250,16 @@ def from_line(
         if req is not None:
             try:
                 req = Requirement(req)
-            except InvalidRequirement:
+            except InvalidRequirement as e:
                 if os.path.sep in req:
                     add_msg = "It looks like a path."
                     add_msg += deduce_helpful_msg(req)
                 elif '=' in req and not any(op in req for op in operators):
                     add_msg = "= is not a valid operator. Did you mean == ?"
                 else:
-                    add_msg = traceback.format_exc()
+                    msg = str(e)
+                    msg = msg.replace('Invalid requirement, ', '')
```

Comment:
This `replace()` seems brittle and hacky to me, but I'll let others weigh in. The `packaging` project probably shouldn't be including this string to begin with since it's redundant with the exception class being `InvalidRequirement`.

Reply:
Agreed I've created a PR against `packaging` fixing that: https://github.com/pypa/packaging/pull/129